game/src/main/scala/io/aigar/game/Resources.scala

```diff
@@ -45,8 +46,29 @@ class ResourceType(grid:Grid, val min: Int, val max: Int, mass: Int, score: Int)
 
   def spawnResources: Unit = {
     val ratio = (positions.length - min).toFloat / (max - min)
-    if (scala.util.Random.nextFloat >= ratio) {
+
+    // Add MinPositiveValue to prevent obtaining zero since ratio can be 0
+    if ((scala.util.Random.nextFloat + scala.Float.MinPositiveValue) >= ratio) {
       positions :::= List(grid.randomPosition)
     }
   }
-}
\ No newline at end of file
+
+  // Small loop to detect if a cell can eat a resource
+  def detectCollision(players: List[Player]): Unit = {
+    for(player <- players) {
+      for(cell <- player.cells) {
+        for(position <- positions){
+          if(cell.contains(position)){
+            reward(cell, mass, score)
+            // Returns a new list without the resource that has been consumed
+            positions = positions.filterNot(a => a == position)
+          }
+        }
+      }
+    }
+  }
+
+  def reward(cell: Cell, mass: Int, score: Int): Unit = {
+    cell.mass += mass
```

Comment:
I actually removed it because it was there first. The score of the player is not used so instead I think I should only send a `scoreMessage` to the `scoreThread` to increase the score on the leaderboard... In other words, I don't think the player needs to know its score. What do you think?